libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/sync/RustSyncService.kt

```diff
@@ -33,19 +36,36 @@ class RustSyncService(
     private val innerSyncService: SyncServiceInterface,
     sessionCoroutineScope: CoroutineScope
 ) : SyncService {
+    private val mutex = Mutex()
+    private val startSyncReasonSet = mutableSetOf<StartSyncReason>()
```

Comment:
I don't know enough about Kotlin, but can the mutex hold the startSyncReasonSet itself, like we'd do in Rust `Mutex<HashSet<StartSyncReason>>`? That would make it more obvious that one needs the mutex to read/write to the set.

Reply:
I don't think we can do something like that, `Mutex` in Kotlin can't hold values.